### PR TITLE
adds storybook changes

### PR DESCRIPTION
**Pull request by juaresba**
Adds the necessary modifications to the template so it comes with Storybooks.js running out of the box.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,5 +58,6 @@
       "index.test.js",
       "lib/**/*.test.js"
     ]
-  }
+  },
+  "sideEffects": false
```
Comment (emersonlaurentino) [inline package.json]: this must be inside the template package

File: template/vtex/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+    stories: ['../src/**/*.stories.tsx'],
```
Comment (emersonlaurentino) [inline template/vtex/.storybook/main.js]: stories must be at the root of the project and not within the `src` folder
```suggestion
    stories: ['../stories/*.stories.tsx'],
```

File: template/vtex/src/stories/0-Welcome.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (emersonlaurentino) [inline template/vtex/src/stories/0-Welcome.stories.tsx]: can u move the folder `stories` to the root folder?

**Review by emersonlaurentino — CHANGES_REQUESTED** [review]
you will need to add theses fields inside of `compilerOptions` on tsconfig.json file
```
  "rootDirs": ["src", "stories"],
  "baseUrl": "src",
```

File: package.json (hunk 1 above)
Comment (juaresba) [reply package.json]: Oh wow I lost attention here, thanks.

**Review by emersonlaurentino — APPROVED** [review]
LGTM

**Pull request merged**